themes/docs-new/layouts/shortcodes/unified_mode.md

```diff
@@ -0,0 +1,192 @@
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  properties.each do |property, value|
+    send(property, value)
+  end
+end
+```
+
+#### Run the Resource Explicitly
+
+Another solution is to continue saving the resource as a variable, declaring `action :nothing` within the codeblock and then explicitly running the action in another codeblock.
+
+The pattern of saving a resource as a variable and then forcing it to run at compile time with an explicit `run_action` works as it has in the past, but it not a preferred pattern. Unified mode forces resource execution to compile time by default, which makes this pattern redundant.
+
+```ruby
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  action :nothing
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+
+# Run the action explicitly
+sr.run_action(:create)
+```
+
+## Actions on Later Resources
+
+Since unified mode executes your resource as it is compiled, you cannot send immediate notifications to resources declared later.
+
+### `:immediate` Notifications to Later Resources
+
+Unified mode delays immediate notifications to later resources.
+In unified mode the Chef Infra Client saves immediate notifications and executes them when the later resource is parsed. Immediate notifications to prior resources and delayed notifications behave the same as they did before unified mode.
+
+Chaining immediate notifications results in the same behavior as before unified mode. Changing the order of resources results does not change the results of a notification chain. Chained notifications in the same order during the Chef Infra Client run and immediately sends a notification to a target resource that was parsed earlier and saves a notification and sends it immediately after it parses a later resource.
+
+The changes to sending immediate notification could result in subtle changes to behaviors in some resources, but it not a breaking change to common patterns of writing resources.
+
+Chaining immediate notifications to later resources:
+
+```ruby
+remote_file "#{Chef::Config[:file_cache_path]}/myservice.tgz" do
+  source "http://acme.com/myservice.tgz"
+  notifies :extract, "archive_file[myservice.tgz]", :immediately
+end
+
+archive_file "#{Chef::Config[:file_cache_path]}/myservice.tgz" do
+  destination '/srv/myservice'
+  notifies :start, "service[myservice]", :immediately
+  action :nothing
+end
+
+service "myservice" do
+  action :nothing
+end
+```
+
+### `:before` Notifications to Later Resources
+
+Unified mode breaks resources with `before` notifications to later resources. In unified mode, you must declare a resource before you can act on it.
```

Comment:
> In unified mode, you must declare a resource before you can act on it.

That's sort of incorrect, you can send delayed notifications to it fine, you can send immediate notifications to it with qualifications, but you just can't do anything with a before notification.

I'm not sure I can rephrase in any way that isn't just redundant with the prior sentence.

```suggestion
In unified mode, you must declare a resource before sending a `before` notification to it.
```